lib/board.rb

```diff
@@ -18,18 +20,26 @@ def valid_coordinate?(coordinate)
     @cells.keys.include?(coordinate)
   end
 
-  def valid_placement?(ship, coordinate)
-    letters = []
-    numbers = []
+  def letters_array(coordinate)
     coordinate.each do |letter|
-      letters << letter[0]
-      numbers << letter[-1]
+      @letters << letter[0]
+    end
+    letters_range = @letters[0]..@letters[-1]
+    letters_range.to_a
+  end
+
+  def numbers_array(coordinate)
+    coordinate.each do |number|
+      @numbers << number[-1]
     end
-    numbers_range = numbers[0]..numbers[-1]
-    letters_range = letters[0]..letters[-1]
-    if letters_range.to_a.uniq.length == 1 && numbers == numbers_range.to_a && coordinate.count == ship.length
+    numbers_range = @numbers[0]..@numbers[-1]
+    numbers_range.to_a
+  end
+
+  def valid_placement?(ship, coordinate)
+    if letters_array(coordinate).uniq.length == 1 && @numbers == numbers_array(coordinate) && coordinate.count == ship.length
       true
-    elsif letters_range.to_a.uniq.length != 1 && letters == letters_range.to_a && numbers.one? && coordinate.count == ship.length
+    elsif letters_array(coordinate).uniq.length != 1 && @letters == letters_array(coordinate) && @numbers.one? && coordinate.count == ship.length
```

Comment:
Thank you for documenting ways we could further refactor. Until then we will need to leave our if statements in tact for our tests to pass. Can you please re-enter them so that I can test that the functionality is working?

Reply:
Thank you for pointing that out! It should pass now.